docs/source/getting_started/getting_started.md

```diff
@@ -423,7 +390,269 @@ curl -X POST "$URL/users/u1234/personalized_documents" \
+
+We made sure that all search APIs in which you can refer to a specific snippet also work with just the document id even if the document has multiple snippets. _But_ the results might not be quite as expected as there can be a huge difference in the search results.
+
+## Snippet Id
+
+Snippets are identified by a snippet id consisting of both the document id and a sub id.
+
+Each document is guaranteed to have a snippet with sub id equals `0`. But besides that
+there are no guarantees about the value of the `sub_id`. E.g. a document might have snippet
+id's with the sub ids `0`, `400`, `2334`.
+
+For example if we ingest a document like
+
+```json
+{
+    "id": "12345",
+    "snippet": "some text"
+},
+```
+
+It would have a single snippet with the id `{ "document_id": "12345", "sub_id": 0 }`.
+
+So if we have following semantic search example
+
+```bash
+curl -X POST "$URL/semantic_search" \
+    --header "authorizationToken: $FRONTOFFICE_TOKEN" \
+    --header "Content-Type: application/json" \
+    --data '{
+        "document": { "id": "12345" },
+        "include_properties": true
+    }'
+```
+
+we could also use the snippet id instead like shown below
+
+```bash
+curl -X POST "$URL/semantic_search" \
+    --header "authorizationToken: $FRONTOFFICE_TOKEN" \
+    --header "Content-Type: application/json" \
+    --data '{
+        "document": { "id": { "document_id": "12345", "sub_id": 0 },
+        "include_properties": true
+    }'
+```
+
+This pattern of replacing document id string with a snippet id struct is also used in all other places
+a snippet id can be used.
+
+Be aware that using `{ "id": "12345" }` represents the whole document but `{ "document_id": "12345", "sub_id": 0 }` represents the first snippet specifically and for some APIs this makes a difference
+for documents which have multiple embeddings as described in the sections below.
+
+## Ingestion
+
+There are two APIs with which you can ingest documents in a way where it can
+produce multiple snippets:
+
+- using the [`split` option with `snippet`](#automatic-document-splitting)
+- using the [`split` option with `file`](#file-upload)
+
+be aware that you need to use sufficient long text for it to be split.
+
+You could use the example from the [`split` option section](#automatic-document-splitting) or use
+your own document, something along ~1000 words will likely generate multiple splits.
+
+## Semantic Search
+
+As mentioned the search results are based on snippets not documents, as such a search
+can yield multiple different snippets of the same document.
+
+The semantic search API can also be used based on a specific snippet (and for systems
+which have documents with multiple embeddings that is recommended):
+
+```bash
+curl -X POST "$URL/semantic_search" \
+    --header "authorizationToken: $FRONTOFFICE_TOKEN" \
+    --header "Content-Type: application/json" \
+    --data '{
+        "document": { "id": { "document_id": "effaffe", "sub_id": 2 },
```

Comment:
We are sure as in this is how it's specified and published.

The idea is to be able to use both interchangeably.

Most times this will be called from js/ts where can just pass a `string| { document_id: string, sub_id: int }` type.

In the result we don't do so because we want to but because we can't just replace `id: string` with `id: {}` without it being a braking change.

Originally when designing there was at some point the misconception that we had named the id field `document_id` like in sql to allow more forward compatibility, but it turned out we did not.

In general I would propose next time we first update the documentation including tutorial like use-cases (weather we publish them or not) as this highlights such discrepancies much more clean.

Through if we still could do braking change wrt. the snippet id API I would do following:

- rename `snippet_id` to `target_id`  and `Snippet` in that context to `SearchTarget`
- *maybe* use `document_id: string` or `target_id: opaque_string` as inputs and outputs, deprecate `id` in both input and output, but also *maybe* not, I'm bit worried about snippet ids (no matter the representation)  because they are mostly non predictable and non query-able, which could be an issue for some customer use-cases

Through AIFIK this wasn't an option already ~4 weeks ago when I first pointed out that the term of snippet and in turn snippet id is sub optimal.